### PR TITLE
Themes: Fix Current Theme Bar for Retired Themes

### LINKED ISSUE #10760: Themes: Current Theme Bar and Theme Sheet Broken for Retired Theme
First reported by @meremagee in Slack a while ago for a site that uses a retired theme:

current-theme-bar-broken

This is a regression stemming from our Redux arch refactor (#8785). We're now always fetching theme data from the generic wpcom themes endpoint (as opposed to the site specific one) in order to not store what are largely identical sets of theme data for each individual site of a user, but that list doesn't include retired themes.

I can see two possible strategies to fix this:
1. Merge data received from the active-theme endpoint into our `themes` Redux tree. That endpoint however doesn't currently include all attributes, so we cannot obtain all data needed e.g. for the theme sheet. We could however extend it to return that data (maybe with a separate `?extended=true` flag or whatever the convention here is).
2. Do a separate per-site query in case `wpcom` doesn't have any data. I wonder where we'd store that one theme then -- I don't think it can go to a site specific Redux subtree, since all other (non-retired) themes will still live in the `wpcom` subtree. So it'd have to go there, too.

Seems like retired theme info should live in the `wpcom` subtree either way then.

I think I'm leaning towards strategy no 1 right now.

### PR DESCRIPTION
**Pull request by ockham**
This PR:
* Changes the  `ACTIVE_THEME_REQUEST_SUCCESS` action to pass an entire theme object instead of just select fields.
* Changes the `requestActiveTheme()` action to use `receiveTheme()` to store that theme object in our Redux tree. Unfortunately, this requires us to use `getState()` in order to find out whether we're dealing with a Jetpack site (in which case the theme object should be stored in the corresponding subtree) or with a WPCOM one (in which case the theme object should go to the generic `wpcom` subtree).
* Removes the `<QueryTheme />` component from `<CurrentTheme />` since it's no longer needed -- the theme object is now fetched by `<QueryActiveTheme />`.
* Turns `<CurrentTheme />` into an ES6 class.

To test:
* Verify that the current theme bar still works for WPCOM and Jetpack sites if the current theme isn't retired.
  * Check that clicking on the buttons triggers corresponding `[Action: Current Theme Info [object Object]]` events (or `Customize`, or `Support` instead of `Info`, respectively).
* Verify that it also works with a retired theme:
  * In order to activate a retired theme, go to your test site's `wp-admin/wp-admin/themes.php?search=Ecto`. Activate the Ecto theme.
  * In Calypso, check the site's Current Theme Bar. Verify that the theme name is displayed, and that the Customize and Info links work. (Note that the theme sheet itself doesn't work yet -- that's for another PR.)

Fixes #10760. (But we should file an issue for theme sheets for retired themes -- see convo in #10760 for a spec.)

### CODE REVIEW

File: client/my-sites/themes/current-theme/index.jsx [hunk 1]
```diff
@@ -34,12 +33,12 @@ const CurrentTheme = React.createClass( {
 		siteId: PropTypes.number.isRequired,
 		// connected props
 		currentTheme: PropTypes.object
-	},
+	}
 
-	trackClick: trackClick.bind( null, 'current theme' ),
+	trackClick = trackClick.bind( null, 'current theme' )
```
Comment (budzanowski) [inline client/my-sites/themes/current-theme/index.jsx]: Cant we ?
`trackClick = event => trackClick( 'current theme', event );`

Comment (ockham) [inline client/my-sites/themes/current-theme/index.jsx]: Yes we can! 😄

**Comment by seear** [comment]
This branch breaks _activate_ for me from the theme list (any theme, any site). Click on _Activate_ in the ... menu and nothing happens. Works fine from the info sheet.

**Comment by budzanowski** [comment]
It also breaks for me. No action is dispatched upon click.

**Comment by ockham** [comment]
Must be the ES6ification. Maybe I'm missing curly braces around the bound `trackClick`...

**Comment by ockham** [comment]
Hah. I must've branched that after I merged #10905 (and before I reverted it). A simple rebase seems to've fixed it.

**Pull request merged**